spec/dfe/analytics/event_matcher_spec.rb

```diff
@@ -0,0 +1,131 @@
+RSpec.describe DfE::Analytics::EventMatcher do
+  subject { described_class.new(event, logging[:event_filters]) }
+
+  context 'when event is a database update' do
+    let(:event) do
+      {
+        'entity_table_name' => 'application_choice_details',
+        'event_type' => 'update_entity',
+        'data' => [
+          { 'key' => 'course_option_id', 'value' => ['12345'] },
+          { 'key' => 'application_form_id', 'value' => ['42'] }
+        ]
+      }
+    end
+
+    describe '.matched?' do
+      context 'when the table name and event type matches' do
+        let(:logging) do
+          {
+            event_filters: [
+              {
+                event_type: 'update_entity',
+                entity_table_name: 'application_choice_details'
```

Comment:
@asatwal I appreciate it's just a test but application_choice_details isn't an entity_table_name in any of our databases - worth changing for clarity?

Reply:
Yes, it's worth changing. I was trying to think of a realistic example, but my application knowledge isn't that great. What would be the entity_table_name for the current problems we are experiencing with the nightly job ?

Reply:
course_options I think

Reply:
Updated to a more realistic example.